Fix enum formatting again

Paxxi opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->
This attempt removes the enums from fmt altogether
as the wrapper functions will convert to int before fmt
sees them.

This makes the changes independent from fmt so we don't
have to worry about different behaviour for different versions.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->
tests pass
## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

Montellese commented on xbmc/utils/StringUtils.h:
```diff
@@ -47,33 +57,16 @@ DEF_TO_STR_VALUE(foo) // outputs "4"
-template<typename T, typename std::enable_if_t<std::is_enum<T>::value, int> = 0>
-std::ostream& operator<<(std::ostream& os, const T& value)
+template<typename T, std::enable_if_t<!std::is_enum<T>::value, int> = 0>
+constexpr auto&& enum_to_int(T&& arg) noexcept
```
Why does this method not follow our naming conventions?

Paxxi replied:
I got sucked in by the type traits naming :)

Montellese reviewed (commented):
Looks good but I would split the PR into three commits:

1. revert the commit from the previous PR to clearly mark that it didn't solve the problem.
2. clang formatting
3. the proper fix commit with `enum_to_int()`

Paxxi committed:
Revert enum fix, didn't work

Paxxi committed:
Fix enum formatting

This attempt removes the enums from fmt altogether, fmt never sees an enum,
only ints.

Paxxi commented:
ok cleaned up

Paxxi commented:
ping @Montellese @MilhouseVH tested it with success, good to merge?

Montellese reviewed (approved):
Sorry I thought I already approved.
